### PR TITLE
Batched uri resolution

### PR DESCRIPTION
**Pull request by jackrobison**
Goes with https://github.com/lbryio/lbryum-server/pull/31

### CODE REVIEW

File: lib/commands.py [hunk 1]
```diff
@@ -925,7 +927,7 @@ def getvalueforname(self, name, raw=False):
             depth = self.network.get_server_height() - height
         result = Commands._verify_proof(name, block_header['claim_trie_root'], response,
                                         height, depth)
-        return self.parse_and_validate_claim_result(result, raw)
+        return self.parse_and_validate_claim_result(result, raw=raw)
```
Comment (lyoshenka) [inline lib/commands.py]: :+1:

File: lib/commands.py [hunk 2]
```diff
@@ -952,8 +954,198 @@ def getdefaultcertificate(self):
...
+            if len(results) and len(results) % page_size == 0:
+                yield results[-page_size:]
+                yielded_page = True
+
+        # if we didn't get a full page of results, yield what results did get
+        if not yielded_page:
+            yield results
+
+    def get_channel_claims_page(self, channel_claim_infos, certificate, page, page_size=10):
+        page = page or 0
```
Comment (lyoshenka) [inline lib/commands.py]: pagination should start on page 1

File: lib/commands.py [hunk 3]
```diff
@@ -952,8 +954,198 @@ def getdefaultcertificate(self):
...
+        page_size = max(page_size, 1)
```
Comment (lyoshenka) [inline lib/commands.py]: should check max page size here?

File: lib/commands.py [hunk 4]
```diff
@@ -952,8 +954,198 @@ def getdefaultcertificate(self):
...
+        elif 'error' not in result:
+            result['error'] = "claim not found"
+            result['success'] = False
+            result['uri'] = str(parsed_uri)
+
+        return result
+
     @command('n')
-    def getvalueforuri(self, uri, raw=False):
+    def getvalueforuri(self, uri, raw=False, page=0, page_size=10):
```
Comment (lyoshenka) [inline lib/commands.py]: this should just call `getvaluesforuris`. DRY!

File: lib/simple_config.py [hunk 5]
```diff
@@ -22,8 +22,9 @@ def set_config(c):
 
 DEFAULT_CONFIG = {
     'default_servers': {
-        'lbryum4.lbry.io': {'t': '50001'},
-        'lbryum5.lbry.io': {'t': '50001'},
+        '54.209.149.242': {'t': '50001'},
+        # 'lbryum4.lbry.io': {'t': '50001'},
```
Comment (lyoshenka) [inline lib/simple_config.py]: intentional??

File: lib/commands.py (hunk 2 above)
Comment (jackrobison) [inline lib/commands.py]: It does, looking up page 0 only returns the page count.

File: lib/commands.py (hunk 3 above)
Comment (jackrobison) [inline lib/commands.py]: Done

File: lib/commands.py (hunk 4 above)
Comment (jackrobison) [inline lib/commands.py]: Done

File: lib/commands.py (hunk 2 above)
Comment (lyoshenka) [inline lib/commands.py]: is that intentional? hidden feature?

**Pull request merged**